siad/apiwallet.go

```diff
@@ -15,8 +15,9 @@ func (d *daemon) walletAddressHandler(w http.ResponseWriter, req *http.Request)
 		return
 	}
 
-	// TODO: Explain what's happening here, and why it doesn't look like all of
-	// the other calls.
+	// Since coinAddress is not a struct, we define one here so that writeJSON
+	// writes an object instead of a bare value. In addition, we transmit the
+	// coinAddress as a hex-encoded string rather than a byte array.
 	writeJSON(w, struct {
 		Address string
 	}{fmt.Sprintf("%x", coinAddress)})
```

Comment:
hmm. Interesting that we would write it as hex through the api. Half of me thinks that the front-end should be what worries about format, but the other half says that defaulting to hex for really simple calls is good.

I guess it's good to leave it for now